weldx/tests/asdf_tests/test_weldx_file.py

```diff
@@ -0,0 +1,307 @@
+"""Tests for the WeldxFile class."""
+import io
+import pathlib
+import tempfile
+from io import BytesIO
+
+import asdf
+import pytest
+
+from scripts import welding_schema
+from weldx import WeldxFile
+from weldx.asdf.util import get_schema_path
+from weldx.types import SupportsFileReadWrite
+
+
+class ReadOnlyFile:
+    """Simulate a read-only file."""
+
+    def __init__(self, tmpdir):
+        """."""
+        fn = tempfile.mktemp(suffix=".asdf", dir=tmpdir)
+        print("fn", fn)
+        with open(fn, "wb") as fh:
+            asdf.AsdfFile(tree=dict(hi="there")).write_to(fh)
+        self.mode = "rb"
+        self.file_read_only = open(fn, mode=self.mode)
+
+    def read(self, *args, **kwargs):
+        """."""
+        return self.file_read_only.read(*args, **kwargs)
+
+    @staticmethod
+    def readable():
+        """."""
+        return True
+
+
+class WritableFile:
+    """Example of a class implementing SupportsFileReadWrite."""
+
+    def __init__(self):
+        """."""
+        self.to_wrap = BytesIO()
+
+    def read(self, *args, **kwargs):
+        """."""
```

Comment:
I guess these are to silence pydocstyle? I would prefer to set the corresponding `# noqa: D...` instead for each function (or at top level in the file/class even if that is supported)